Formula/texlive.rb

```diff
@@ -0,0 +1,499 @@
+
+    tex_resources = ["texlive-extra", "install-tl", "texlive-texmf"]
+    python_resources = ["Pygments"]
+
+    resources.each do |r|
+      r.stage do
+        next if tex_resources.include? r.name
+
+        next if python_resources.include? r.name
+
+        if File.exist? "Makefile.PL"
+          system "perl", "Makefile.PL", "INSTALL_BASE=#{libexec}",
+            "CCFLAGS=-I#{Formula["freetype"].opt_include}/freetype2"
+          system "make"
+          system "make", "install"
+        else
+          system "perl", "Build.PL", "--install_base", libexec
+          system "./Build"
+          system "./Build", "install"
+        end
+      end
+    end
+
+    # Install Python resources
+    venv = virtualenv_create(libexec, "python3")
+    venv.pip_install resource("Pygments")
+
+    # Install TeXLive resources
+    resource("texlive-extra").stage do
+      share.install "tlpkg"
+    end
+
+    resource("install-tl").stage do
+      cd "tlpkg" do
+        (share/"tlpkg").install "installer"
+        (share/"tlpkg").install "tltcl"
+      end
+    end
+
+    resource("texlive-texmf").stage do
+      share.install "texmf-dist"
+    end
+
+    # Clean unused files
+    rm_rf share/"texmf-dist/doc"
+    rm_rf share/"tlpkg/installer/wget"
+    rm_rf share/"tlpkg/installer/xz"
+
+    # Set up config files to use the correct path for the TeXLive root
+    inreplace buildpath/"texk/kpathsea/texmf.cnf",
+      "TEXMFROOT = $SELFAUTOPARENT", "TEXMFROOT = $SELFAUTODIR/share"
+    inreplace share/"texmf-dist/web2c/texmfcnf.lua",
+      "selfautoparent:texmf", "selfautodir:share/texmf"
+
+    # Fix path resolution in some scripts (these should be upstreamed)
+    pathfix_files = [
+      buildpath/"texk/texlive/linked_scripts/cjk-gs-integrate/cjk-gs-integrate.pl",
+      buildpath/"texk/texlive/linked_scripts/crossrefware/bbl2bib.pl",
+      buildpath/"texk/texlive/linked_scripts/crossrefware/bibdoiadd.pl",
+      buildpath/"texk/texlive/linked_scripts/crossrefware/bibmradd.pl",
+      buildpath/"texk/texlive/linked_scripts/crossrefware/biburl2doi.pl",
+      buildpath/"texk/texlive/linked_scripts/crossrefware/bibzbladd.pl",
+      buildpath/"texk/texlive/linked_scripts/crossrefware/ltx2crossrefxml.pl",
+      buildpath/"texk/texlive/linked_scripts/texlive/tlmgr.pl",
+      buildpath/"texk/texlive/linked_scripts/pedigree-perl/pedigree.pl",
+      buildpath/"texk/texlive/linked_scripts/tlshell/tlshell.tcl",
+      share/"tlpkg/TeXLive/TLUtils.pm",
+    ]
+
+    inreplace pathfix_files, "SELFAUTOPARENT", "TEXMFROOT"
+
+    args = std_configure_args + [
+      "--disable-dvisvgm", # needs its own formula
+      "--disable-missing",
+      "--disable-native-texlive-build", # needed when doing a distro build
+      "--disable-ps2eps", # provided by ps2eps formula
+      "--disable-psutils", # provided by psutils formula
+      "--disable-static",
+      "--disable-t1utils", # provided by t1utils formula
```

Comment:
Can we group the existing formula.

Reply:
Yes, I'll change this as it will look better.